modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -1597,14 +1595,8 @@ void cv::gimpl::GStreamingExecutor::setSource(GRunArgs &&ins)
         island_meta_info = GIslandModel::traceIslandName(op.nh, m_gim);
 #endif // OPENCV_WITH_ITT
 
-        // If Island Executable is recompiled, all its stuff including internal kernel states
-        // are recreated and re-initialized automatically.
-        // But if not, we should notify Island Executable about new started stream to let it update
-        // its internal variables.
-        if (!islandsRecompiled)
-        {
-            op.isl_exec->handleNewStream();
-        }
+        // Notify island executable about new coming stream to let it update its internal variables.
+        op.isl_exec->handleNewStream();
```

Comment:
So according to the old code and comment this method shouldn't be called on any SetSource.
What's the right behavior here?

Reply:
Thanks for the question! Behavior is explained in the PR description now! (For addressing your comment about lack of explanation of what is going in this PR)